api/src/common/utils.rs

```diff
@@ -23,29 +28,90 @@ pub const fn near_data_to_near_token(data: near_api_types::Data<u128>) -> NearTo
 // We default to false as we can't know if an error is critical or not without the types
 // so to keep it safe it's better to retry
 
-pub fn is_critical_blocks_error(err: &RpcError) -> bool {
-    is_critical_json_rpc_error(err, |_| false)
+pub fn to_retry_error<T, E: std::fmt::Debug + Send + Sync>(
+    err: SendRequestError<E>,
+    is_critical_t: impl Fn(&SendRequestError<E>) -> bool,
+) -> RetryResponse<T, SendRequestError<E>> {
+    if is_critical_t(&err) {
+        RetryResponse::Critical(err)
+    } else {
+        RetryResponse::Retry(err)
+    }
 }
 
-pub fn is_critical_validator_error(err: &RpcError) -> bool {
-    is_critical_json_rpc_error(err, |_| false)
+pub fn is_critical_blocks_error(err: &SendRequestError<RpcBlockError>) -> bool {
+    is_critical_json_rpc_error(err, |err| match err {
+        RpcBlockError::UnknownBlock { .. }
+        | RpcBlockError::NotSyncedYet
+        | RpcBlockError::InternalError { .. } => true,
```

Comment:
Why don't you want to retry this error? If it's not the type you know in advance, you can retry and maybe get correct answer. It was retried before. Let's retry all the unknown InternalErrors